core/solver/multigrid.cpp

```diff
@@ -467,6 +468,118 @@ void MultigridState::run_cycle(multigrid::cycle cycle, size_type level,
 }  // namespace multigrid
 
 
+std::function<size_type(const size_type, const LinOp*)> get_selector(
+    std::string key)
+{
+    static std::map<std::string,
+                    std::function<size_type(const size_type, const LinOp*)>>
+        selector_map{
+            {{"first_for_top", [](const size_type level, const LinOp*) {
+                  return (level == 0) ? 0 : 1;
+              }}}};
+    return selector_map.at(key);
+}
+
+
+typename Multigrid::parameters_type Multigrid::parse(
+    const config::pnode& config, const config::registry& context,
+    const config::type_descriptor& td_for_child)
+{
+    auto factory = Multigrid::build();
+
+    if (auto& obj = config.get("criteria")) {
+        factory.with_criteria(
+            gko::config::get_factory_vector<const stop::CriterionFactory>(
+                obj, context, td_for_child));
+    }
+    if (auto& obj = config.get("mg_level")) {
+        factory.with_mg_level(
+            gko::config::get_factory_vector<const gko::LinOpFactory>(
+                obj, context, td_for_child));
```

Comment:
By default, if user gives the enough mg_level, which less than the generated levels (or limited max_mg_level).
Multigrid will use each for each level

Reply:
I would still advocate to only allow a single value here. How many mg_levels users need to provide should again depend on the matrix, so it would be unknowable for the configuration.
Also going to multiple levels later would be backwards compatible, so there would be no issue.
(same goes for the other vector parameters here)